src/Plugin/Field/FieldType/MinisiteItem.php

```diff
@@ -165,15 +160,20 @@ public function fieldSettingsForm(array $form, FormStateInterface $form_state) {
   public function preSave() {
     parent::preSave();
 
-    // Minisite presave.
-    $this->asset_path = Minisite::preSave($this->entity, $this->getEntity());
```

Comment:
This was one of the 'logic leaking' places. `Minisite` is/was not an entity, so it does not know anything about saving or pre-saving.

Changed this now that a Minisite instance is instantiated and a primary entry point (index file) is retrieved.